linux/check_os_fullname/check_os_fullname.yml

```diff
@@ -95,11 +86,37 @@
               {%- if vm_guest_id is match('freebsd.*') -%}otherGuestFamily
               {%- else -%}linuxGuest{%- endif -%}
 
-        - name: "Validate guest id, full name and family in guest info"
+        - name: "Validate guest id, full name and family in guest info for compatible hardware versions"
           include_tasks: validate_os_fullname.yml
+          with_items: "{{ compatible_hardware_versions }}"
+          loop_control:
+            loop_var: compatible_hw_version
+
       rescue:
         - name: "Test rescue"
           include_tasks: ../../common/test_rescue.yml
       always:
         - name: "Collect VMware Tools logs"
           include_tasks: ../utils/collect_vmtools_logs.yml
+
+        - name: "Revert to base snapshot to downgrade VM hardware version and update VM info"
+          when:
+            - vm_base_hardware_version | default('')
+            - vm_current_hw_version | default('')
+            - vm_current_hw_version | int > vm_base_hardware_version | int
+          block:
+            - name: "Revert to base snapshot"
+              include_tasks: ../../common/vm_revert_snapshot.yml
+              vars:
+                snapshot_name: "{{ base_snapshot_name }}"
+
+            # FreeBSD VM's base snapshot is in powered off state
```

Comment:
After reverting to base snapshot, all the guest types can update inventory.

Reply:
FreeBSD's snapshot is in power-off state, so here uses update_inventory.yml to power on it and wait for vmtools running. For other Linux, it is not necessary as the following tasks only get VM info and guest info, so that the results.log can show VM and guest info as VM before upgrading hardware versions.